### PR TITLE
Select: Add orange indicator to selected item

### LINKED ISSUE #79573: [accessibility issue] Focus states for the Select component don't meet 3:1 contrast requirement
**Steps to reproduce**:
Go to any Select component (or on the docs website: https://developers.grafana.com/ui/latest/index.html?path=/story/forms-select--multi-plain-value) in light or dark mode. Navigate the combobox to initiate a focus state which highlights the item to be selected. The focus state between the highlighted item next to an unhighlighted item is:

- 1:42 in dark mode (https://www.siegemedia.com/contrast-ratio#rgb%2847%2048%2054%29-on-%23111217) -- note that as the selected state has an opacity you have to color pick it to rgb(47 48 54) to get a proper reading.
- 1.26 in light mode (https://www.siegemedia.com/contrast-ratio#rgb%28228%20229%20230%29-on-%23fff) -- note that as the selected state has an opaicty you have to color pick it to rgb(228 229 230) to get a proper reading. 

**Expected result**
Color contrast is at least 3:1 as per [WCAG 2.4.13](https://www.w3.org/WAI/WCAG22/Understanding/focus-appearance.html) and [WCAG 1.4.11](https://www.w3.org/WAI/WCAG22/Understanding/non-text-contrast.html).

Screenshot of Select component in light mode on the component documentation website.

Screenshot of Select component in dark mode on the component documentation website.

### PR DESCRIPTION
**Pull request by tskarhed**
**What is this feature?**

- Adds a "pill" to the side of the selected option
- Adds underline text style to focus and hovered options

image

Considerations:
1. Selected and focus are the same when initially opened. Does it look to cluttered?
2. Both focused and hovered can be present at the same time, causing a duplicate look

image

**Why do we need this feature?**

More accessibility

**Who is this feature for?**

Everbody

**Which issue(s) does this PR fix?**:

Fixes #79573

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: packages/grafana-ui/src/components/Select/getSelectStyles.ts [hunk 1]
```diff
@@ -27,6 +28,7 @@ export const getSelectStyles = stylesFactory((theme: GrafanaTheme2) => {
 
       '&:hover': {
         background: theme.colors.action.hover,
+        '& span': { textDecoration: 'underline' },
```
Comment (tskarhed) [inline packages/grafana-ui/src/components/Select/getSelectStyles.ts:31]: We could consider removing this to differentiate between focused and hovered

Comment (Clarity-89) [reply packages/grafana-ui/src/components/Select/getSelectStyles.ts:31]: I think it looks pretty clear without the underline imo, having the underline might imply to some that an option is a link.

**Review by Clarity-89 — APPROVED** [review]
I think this looks great 🚀  Left a minor comment.

File: packages/grafana-ui/src/components/Select/getSelectStyles.ts (hunk 1 above)
Comment (ashharrison90) [reply packages/grafana-ui/src/components/Select/getSelectStyles.ts:31]: big +1 to this, i don't think we should underline

File: packages/grafana-ui/src/components/Select/getSelectStyles.ts [hunk 2]
```diff
@@ -58,12 +59,24 @@ export const getSelectStyles = stylesFactory((theme: GrafanaTheme2) => {
     optionFocused: css({
       label: 'grafana-select-option-focused',
       background: theme.colors.action.focus,
+      '& span': { textDecoration: 'underline' },
```
Comment (ashharrison90) [inline packages/grafana-ui/src/components/Select/getSelectStyles.ts:62]: i don't think we should underline here either btw

**Comment by diegommm** [comment]
Hi! Greetings from the Release Guild. Just letting you know I'm updating your PR's milestone to [11.2.x](https://github.com/grafana/grafana/milestone/529). If you need to see your changes in a previous version, please consider adding a backport label.

Thank you!

**Pull request merged**